alibi_detect/utils/loading.py

```
@@ -0,0 +1,625 @@
+        The backend.
+
+    Returns
+    -------
+    The loaded embedding.
+    """
+    src = cfg['src']
+    layers = cfg['layers']
+    typ = cfg['type']
+    if backend == 'tensorflow':
+        emb = load_embedding_tf(src, embedding_type=typ, layers=layers)
+    else:
+        emb = None
+    return emb
+
+
+def _load_tokenizer(cfg: dict) -> AutoTokenizer:
+    """
+    Loads a text tokenizer from a tokenizer config dict.
+
+    Parameters
+    ----------
+    cfg
+        A tokenizer config dict. (see the pydantic schemas).
+
+    Returns
+    -------
+    The loaded tokenizer.
+    """
+    src = cfg['src']
+    kwargs = cfg['kwargs']
+    src = Path(src)
+    tokenizer = AutoTokenizer.from_pretrained(src, **kwargs)
+    return tokenizer
+
+
+def _load_optimizer(cfg: dict,
+                    backend: str) -> Union['tf.keras.optimizers.Optimizer', Callable]:
+    """
+    Loads an optimzier from an optimizer config dict. When backend='tensorflow', the config dict should be in
+    the format given by tf.keras.optimizers.serialize().
+
+    Parameters
+    ----------
+    cfg
+        The optimizer config dict.
+    backend
+        The backend.
+
+    Returns
+    -------
+    The loaded optimizer.
+    """
+    if backend == 'tensorflow':
+        optimizer = load_optimizer_tf(cfg)
+    else:
+        raise NotImplementedError('Loading of pytorch optimizers not currently supported')
+    return optimizer
+
+
+def _get_nested_value(dic: dict, keys: list) -> Any:
+    """
+    Get a value from a nested dictionary.
+
+    Parameters
+    ----------
+    dic
+        The dictionary.
+    keys
+        List of keys to "walk" to nested value.
+        For example, to extract the value `dic['key1']['key2']['key3']`, set `keys = ['key1', 'key2', 'key3']`.
+
+    Returns
+    -------
+    The nested value specified by `keys`.
+    """
+    for key in keys:
+        try:
+            dic = dic[key]
+        except (TypeError, KeyError):
+            return None
```

Is there any risk in ignoring this error?

---

Sorry, can you clarify what you mean please?

---

Ah, ignore me! I didn't check properly how you were using it.